src/server/cea_608_encoder/cea_608_utility.py

```diff
@@ -56,19 +56,67 @@ def bytes_to_byte_pairs(byte_list: list) -> list:
     return byte_pairs
 
 
-def create_byte_pairs(caption_string: str) -> list:
+def which_char_set(caption_char: str) -> str:
+    """Finds which character set the letter is in
+
+    :param caption_char:
+    :return: the name of the char set the caption_char is in
+    """
+    for set_name,set in char_sets.items():
+        for letter,decimal in set.items():
```

Comment:
Instead of double nested loop i think we can do 
`for char_set_name, list_of_characters in char_sets.item():
    if caption_char in list_of_characters:
      return char_set_name`